python/cudf/cudf/testing/_utils.py

```diff
@@ -368,3 +369,23 @@ def sv_to_udf_str_testing_lowering(context, builder, sig, args):
     return cast_string_view_to_udf_string(
         context, builder, sig.args[0], sig.return_type, args[0]
     )
+
+
+class cudf_timeout:
```

Comment:
Is it worth it using something like `@pytest.mark.timeout(...)` from [pytest-timeout](https://github.com/pytest-dev/pytest-timeout) instead of this class?

Reply:
I tried this first, but `pd.CategoricalIndex` is also very slow. So we don't want to timeout for slowness there.

Reply:
Ah okay. Do you think we should just supply the value of `expected_repr` ahead of time? Like
```
expected_repr="CategoricalIndex([       0,        1,        2,        3,        4,        5,\n                         6,        7,        8,        9,\n                  ...\n                  99999990, 99999991, 99999992, 99999993, 99999994, 99999995,\n                  99999996, 99999997, 99999998, 99999999],\n                 categories=[0, 1, 2, 3, ..., 99999996, 99999997, 99999998, 99999999], ordered=False, dtype='category', length=100000000)"
```

Reply:
> Ah okay. Do you think we should just supply the value of `expected_repr` ahead of time? Like
> 
> ```
> expected_repr="CategoricalIndex([       0,        1,        2,        3,        4,        5,\n                         6,        7,        8,        9,\n                  ...\n                  99999990, 99999991, 99999992, 99999993, 99999994, 99999995,\n                  99999996, 99999997, 99999998, 99999999],\n                 categories=[0, 1, 2, 3, ..., 99999996, 99999997, 99999998, 99999999], ordered=False, dtype='category', length=100000000)"
> ```

That would easily break if display options on the machine this pytest runs change. The repr is tightly coupled with quite a few pandas display options.